src/Raven.Server/ServerWide/Maintenance/Sharding/OrchestratorTopologyUpdater.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.Client.ServerWide;
+using Raven.Server.Config.Categories;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide.Commands;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Utils;
+
+namespace Raven.Server.ServerWide.Maintenance.Sharding
+{
+    internal class OrchestratorTopologyUpdater : DatabaseTopologyUpdater
+    {
+        public OrchestratorTopologyUpdater(ServerStore server,
+            RachisConsensus<ClusterStateMachine> engine,
+            ClusterConfiguration clusterConfiguration,
+            DateTime clusterObserverStartTime,
+            ObserverLogger logger) : base(server, engine, clusterConfiguration, clusterObserverStartTime, logger)
+        {
+        }
+
+        protected override (bool Promote, string UpdateTopologyReason) TryPromote(ClusterOperationContext context, ClusterObserver.DatabaseObservationState state, string promotable, ClusterNodeStatusReport nodeStats)
+        {
+            if (IsLastCommittedIndexCaughtUp(context, promotable, state.DatabaseTopology, nodeStats, state.ObserverIteration))
+            {
+                state.DatabaseTopology.PromotablesStatus.Remove(promotable);
+                state.DatabaseTopology.DemotionReasons.Remove(promotable);
+                return (true, $"Node {promotable} is ready to be promoted to orchestrator");
+            }
+
+            return (false, $"Node {promotable} is not ready to be promoted to orchestrator because its index is not caught up yet");
+        }
+
+        private bool IsLastCommittedIndexCaughtUp(ClusterOperationContext context, string node, DatabaseTopology topology, ClusterNodeStatusReport nodeStats, long iteration)
+        {
+            var lastCommittedIndex = _engine.GetLastCommitIndex(context);
+
+            //heartbeat legacy
+            if (nodeStats.ServerReport.LastCommittedIndex == null)
+            {
+                DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Pawel, DevelopmentHelper.Severity.Normal, "Should handle mixed cluster of 5.x and 6?");
```

Comment:
I think we don't 
cc @ppekrol

Reply:
I do not think so, this is also Orchestrator updater so it should be impossible?